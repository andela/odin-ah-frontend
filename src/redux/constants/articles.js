export const SENDING_REQUEST = 'SENDING_REQUEST';
export const CREATE_ARTICLE_SUCCESS = 'CREATE_ARTICLE_SUCCESS';
export const CREATE_ARTICLE_ERROR = 'CREATE_ARTICLE_ERROR';
export const OPEN_PUBLISH_MODAL = 'OPEN_PUBLISH_MODAL';
export const CLOSE_PUBLISH_MODAL = 'CLOSE_PUBLISH_MODAL';
export const SHOW_PUBLISH_ERROR = 'SHOW_PUBLISH_ERROR';
export const HIDE_PUBLISH_ERROR = 'HIDE_PUBLISH_ERROR';
export const HIDE_PUBLISH_RESPONSE = 'HIDE_PUBLISH_RESPONSE';

export const GET_ARTICLE_SUCCESS = 'GET_ARTICLE_SUCCESS';
export const GET_ARTICLE_ERROR = 'GET_ARTICLE_ERROR';
export const UPDATE_ARTICLE_SUCCESS = 'UPDATE_ARTICLE_SUCCESS';
export const UPDATE_ARTICLE_ERROR = 'UPDATE_ARTICLE_ERROR';
