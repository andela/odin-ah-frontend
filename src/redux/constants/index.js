export const LOGIN_USER = 'LOGIN_USER';
export const LOGIN_USER_ERROR = 'LOGIN_USER_ERROR';

export const AUTHENTICATING = 'AUTHENTICATING';

export const LOGOUT_USER = 'LOGOUT_USER';

export const FETCH_PROFILE_BEGIN = 'FETCH_PROFILE_BEGIN';
export const FETCH_PROFILE_SUCCESS = 'FETCH_PROFILE_SUCCESS';
export const FETCH_PROFILE_FAILURE = 'FETCH_PROFILE_FAILURE';

export const SAVE_PROFILE_BEGIN = 'SAVE_PROFILE_BEGIN';
export const SAVE_PROFILE_SUCCESS = 'SAVE_PROFILE_SUCCESS';
export const SAVE_PROFILE_FAILURE = 'SAVE_PROFILE_FAILURE';
