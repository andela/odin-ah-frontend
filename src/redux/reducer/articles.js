import {
  CREATE_ARTICLE_REQUEST,
  CREATE_ARTICLE_SUCCESS,
  CREATE_ARTICLE_ERROR,
  OPEN_PUBLISH_MODAL,
  CLOSE_PUBLISH_MODAL,
  SHOW_PUBLISH_ERROR,
  HIDE_PUBLISH_ERROR
} from '../constants/articles';

const defaultState = {
  open: false,
  errors: {}
};
const articleReducer = (state = defaultState, action) => {
  const {
    type, loading, response, errors
  } = action;
  switch (type) {
    case OPEN_PUBLISH_MODAL:
      return {
        ...state,
        open: true
      };
    case CLOSE_PUBLISH_MODAL:
      return {
        ...state,
        open: false
      };
    case SHOW_PUBLISH_ERROR:
      return {
        ...state, errors
      };
    case HIDE_PUBLISH_ERROR:
      return {
        ...state, errors: {}
      };
    case CREATE_ARTICLE_REQUEST:
      return {
        ...state, loading, response, errors: {}
      };
    case CREATE_ARTICLE_SUCCESS:
      return {
        ...state, loading, open: false, response
      };
    case CREATE_ARTICLE_ERROR:
      return {
        ...state, loading, errors: response.data
      };
    default:
      return state;
  }
};

export default articleReducer;
