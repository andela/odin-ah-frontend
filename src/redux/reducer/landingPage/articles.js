import {
  FETCH_ARTICLES_SUCCESS,
  FETCH_ARTICLES_ERROR,
  FETCH_ARTICLES_LOADING,
  FETCH_ARTICLES_FROM_STORE,
  INVALID_ACTION
} from '../../actions/landingPage/articles';

const initialState = {
  articles: [],
  loadingArticles: false,
  articlesError: false,
  articlesErrorMessage: '',
  articlesInView: [],
  totalPages: null,
  currentPage: null,
  size: 20,
  pagesFetched: 0,
  totalCount: 0
};

const landingPageArticles = (state = initialState, action = {}) => {
  switch (action.type) {
    case FETCH_ARTICLES_SUCCESS: {
      const noPageChange = action.payload.data.page <= state.currentPage;
      const fetchedArticles = action.payload.data.articles;
      const articles = noPageChange ? fetchedArticles : state.articles.concat(fetchedArticles);
      const pagesFetched = noPageChange ? state.pagesFetched : state.pagesFetched + 1;
      const { size } = action.payload.data;
      const sliceStart = (action.payload.data.page - 1) * 20;
      const sliceEnd = action.payload.data.page * 20;
      return {
        ...state,
        articles,
        currentPage: action.payload.data.page,
        totalPages: action.payload.data.totalPages,
        totalCount: action.payload.data.total,
        size,
        pagesFetched,
        loadingArticles: false,
        articlesError: false,
        articlesInView: articles.slice(sliceStart, sliceEnd)
      };
    }
    case FETCH_ARTICLES_ERROR:
      return {
        ...state,
        loadingArticles: false,
        articlesError: true,
        articlesErrorMessage: action.payload
      };
    case FETCH_ARTICLES_LOADING:
      return {
        ...state,
        loadingArticles: true,
        articlesError: false
      };
    case FETCH_ARTICLES_FROM_STORE: {
      const { articles } = state;
      const sliceStart = (action.pageToLoad - 1) * 20;
      const sliceEnd = action.pageToLoad * 20;
      return {
        ...state,
        currentPage: action.pageToLoad,
        loadingArticles: false,
        articlesError: false,
        articlesInView: articles.slice(sliceStart, sliceEnd)
      };
    }
    case INVALID_ACTION:
      return state;
    default:
      return state;
  }
};

export default landingPageArticles;
