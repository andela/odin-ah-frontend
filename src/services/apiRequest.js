import Axios from 'axios';
import { logout } from '../redux/actions/auth/login';

const BASE_API_URL = process.env.REACT_APP_BASE_API_URL;

export class ApiRequest {
  /**
   *
   * @param {'facebook'|'twitter'|'google'} social
   * @return {string}
   */
  static socialAuthUrl(social) {
    return `${BASE_API_URL}/auth/${social}`;
  }

  constructor() {
    this.axios = Axios.create({
      baseURL: BASE_API_URL,
    });
    const token = localStorage.getItem('jwtToken');
    this.setToken(token);
  }

  getInstance() {
    return this.axios;
  }

  setToken(token) {
    if (token) {
      this.axios.defaults.headers.common.Authorization = `Bearer ${token}`;
    } else {
      delete this.axios.defaults.headers.common.Authorization;
    }
  }

  /**
   *
   * @param {{email:string,password:string,username:string}} user
   * @return {Promise<object>} return a Promise of the request
   */
  createUser(user) {
    return this.axios.post('/auth/signup', user);
  }

  /**
   *
   * @param { string } token
   * @return {Promise<any>} 59c3d95d0e458f279d365c14fdcdd1b2305c3696
   */
  verityConfirmationToken(token) {
    return this.axios.get(`/auth/confirmation/${token}`);
  }

  /**
   *
   * @param {{email: string}} data
   * @return {Promise<any>}
   */
  resendVerificationToken(data) {
    return this.axios.post('/auth/confirmation', data);
  }

  loginUser(data) {
    return this.axios.post('/auth/login', data);
  }

  authenticateUser() {
    return this.axios.get('/users');
  }

  registerInterceptors(store = null) {
    this.axios.interceptors.response.use(
      response => response,
      (error) => {
        const { status } = error.response;
        if (status === 401) {
          if (store) {
            store.dispatch(logout);
          }
        }
        return Promise.reject(error);
      }
    );
  }

  createArticle(data) {
    return this.axios.post('/articles', data);
  }

  getArticle(slug) {
    return this.axios.get(`/articles/${slug}`);
  }

  updateArticle(slug, payload) {
    return this.axios.put(`/articles/${slug}`, payload);
  }
}

const apiRequest = new ApiRequest();
export default apiRequest;
