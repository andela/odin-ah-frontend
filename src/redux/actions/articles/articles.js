import apiRequest from '../../../services/apiRequest';

import {
  SENDING_REQUEST,
  CREATE_ARTICLE_SUCCESS,
  CREATE_ARTICLE_ERROR,
  UPDATE_ARTICLE_SUCCESS,
  UPDATE_ARTICLE_ERROR,
  OPEN_PUBLISH_MODAL,
  CLOSE_PUBLISH_MODAL,
  SHOW_PUBLISH_ERROR,
  HIDE_PUBLISH_ERROR,
  HIDE_PUBLISH_RESPONSE,
  GET_ARTICLE_SUCCESS,
  GET_ARTICLE_ERROR
} from '../../constants/articles';

export const openPublishModal = () => ({
  type: OPEN_PUBLISH_MODAL
});
export const closePublishModal = () => ({
  type: CLOSE_PUBLISH_MODAL
});
export const showCreateError = errors => ({
  type: SHOW_PUBLISH_ERROR,
  errors
});
export const hideCreateError = () => ({
  type: HIDE_PUBLISH_ERROR
});
export const hideCreateResponse = () => ({
  type: HIDE_PUBLISH_RESPONSE
});
export const createArticleRequest = articleData => (dispatch) => {
  dispatch({
    type: SENDING_REQUEST
  });
  return apiRequest.createArticle(articleData).then((res) => {
    const { status: statusCode } = res;
    const responseMessage = statusCode === 201 ? 'Article has been published successfully!' : res.data.message;
    dispatch({
      type: CREATE_ARTICLE_SUCCESS,
      response: responseMessage
    });
    return statusCode;
  }).catch((err) => {
    const { response } = err;
    const { status: statusCode } = response;
    dispatch({
      type: CREATE_ARTICLE_ERROR,
      response: err.response.data
    });
    return statusCode;
  });
};


export const getArticleForUpdate = slug => (dispatch) => {
  // change to sending_request
  dispatch({
    type: SENDING_REQUEST
  });
  return apiRequest.getArticle(slug).then((res) => {
    dispatch({
      type: GET_ARTICLE_SUCCESS,
      response: res.data
    });
  }).catch((err) => {
    const { response } = err;
    dispatch({
      type: GET_ARTICLE_ERROR,
      response: response.data,
      statusCode: response.status
    });
  });
};

export const updateArticleRequest = (slug, articleData) => (dispatch) => {
  dispatch({
    type: SENDING_REQUEST
  });
  return apiRequest.updateArticle(slug, articleData).then((res) => {
    const { status: statusCode } = res;
    const responseMessage = statusCode === 200 ? 'Article has been updated successfully!' : res.data.message;
    dispatch({
      type: UPDATE_ARTICLE_SUCCESS,
      response: responseMessage
    });
    return statusCode;
  }).catch((err) => {
    const { response } = err;
    const { status: statusCode } = response;
    dispatch({
      type: UPDATE_ARTICLE_ERROR,
      response: err.response.data
    });
    return statusCode;
  });
};
