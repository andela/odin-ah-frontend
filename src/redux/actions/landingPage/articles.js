import apiRequest from '../../../services/apiRequest';
import { showAlert } from '../notification';

export const FETCH_ARTICLES = 'FETCH_ARTICLES';
export const FETCH_ARTICLES_SUCCESS = 'FETCH_ARTICLES_SUCCESS';
export const FETCH_ARTICLES_ERROR = 'FETCH_ARTICLES_ERROR';
export const FETCH_ARTICLES_LOADING = 'FETCH_ARTICLES_LOADING';
export const FETCH_ARTICLES_FROM_STORE = 'FETCH_ARTICLES_FROM_STORE';
export const INVALID_ACTION = 'INVALID_ACTION';

export const fetchArticlesFailed = error => ({
  type: FETCH_ARTICLES_ERROR,
  payload: error
});

export const fetchArticlesSucceeded = data => ({
  type: FETCH_ARTICLES_SUCCESS,
  payload: data
});

export const fetchArticlesStarted = () => ({
  type: FETCH_ARTICLES_LOADING
});

export const invalidAction = () => ({
  type: INVALID_ACTION
});

export const fetchArticlesFromStore = (pageToLoad, size) => ({
  type: FETCH_ARTICLES_FROM_STORE,
  size,
  pageToLoad
});

export const fetchArticles = (pageToLoad, size) => async (dispatch) => {
  try {
    dispatch(fetchArticlesStarted());
    const result = await apiRequest.fetchArticles(pageToLoad, size);
    const { data } = result;
    dispatch(fetchArticlesSucceeded(data));
  } catch (err) {
    if (err.response) {
      dispatch(fetchArticlesFailed(err.response.data));
      dispatch(showAlert({ type: 'error', text: err.response.data.message }));
    } else if (err.request) {
      dispatch(
        showAlert({
          type: 'error',
          text: 'Could not connect to server. Please check your connection'
        })
      );
    } else {
      dispatch(showAlert({ type: 'error', text: 'An error occurred. Please try again' }));
    }
  }
};

export const fetchArticlePage = page => (dispatch, getState) => {
  if (![1, -1].includes(page)) return dispatch(invalidAction());
  const {
    currentPage, size, pagesFetched, totalPages
  } = getState().landingPageArticles;
  const pageToLoad = currentPage + page;
  if (pageToLoad > totalPages || pageToLoad < 1) {
    return dispatch(invalidAction());
  }
  if (pageToLoad > pagesFetched && pagesFetched < totalPages) {
    return dispatch(fetchArticles(pageToLoad, size));
  }
  if (pageToLoad <= pagesFetched && pagesFetched <= totalPages) {
    return dispatch(fetchArticlesFromStore(pageToLoad, size));
  }
  return dispatch(invalidAction());
};
