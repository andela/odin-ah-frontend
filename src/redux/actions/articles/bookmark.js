import * as actionTypes from '../../constants/articles';
import apiRequest from '../../../services/apiRequest';
import { dispatchError, getErrorMessage, dispatchSuccess } from '../auth/register';


export const fetchBookmarkBegin = () => ({
  type: actionTypes.FETCH_BOOKMARK_ARTICLE_BEGIN,
});

export const fetchBookmarkSuccess = data => ({
  type: actionTypes.FETCH_BOOKMARK_ARTICLE_SUCCESS,
  data,
});

export const fetchBookmarkFailure = error => ({
  type: actionTypes.FETCH_BOOKMARK_ARTICLE_FAILURE,
  error,
});

export const removeBookmarkBegin = () => ({
  type: actionTypes.REMOVE_BOOKMARK_ARTICLE_BEGIN,
});

export const removeBookmarkSuccess = data => ({
  type: actionTypes.REMOVE_BOOKMARK_ARTICLE_SUCCESS,
  data,
});

export const removeBookmarkFailure = error => ({
  type: actionTypes.REMOVE_BOOKMARK_ARTICLE_FAILURE,
  error,
});

export const invalidAction = () => ({
  type: actionTypes.INVALID_ACTION
});

export const fetchArticlesFromStore = (pageToLoad, size) => ({
  type: actionTypes.FETCH_ARTICLES_FROM_STORE,
  size,
  pageToLoad
});


export const fetchBookmarkArticle = (pageToLoad, size) => async (dispatch) => {
  dispatch(fetchBookmarkBegin());
  try {
    const { data } = await apiRequest.getBookmarkArticle(pageToLoad, size);
    dispatch(fetchBookmarkSuccess(data.data));
  } catch (e) {
    dispatch(fetchBookmarkFailure(e.response));
    const { message, type } = getErrorMessage(e);
    dispatchError(message, type, dispatch);
  }
};


export const removeBookmarkArticle = slug => async (dispatch) => {
  dispatch(removeBookmarkBegin());
  try {
    const { data } = await apiRequest.removeBookmarkArticle(slug);
    dispatch(removeBookmarkSuccess(data));
    dispatchSuccess(data.message, 'alert', dispatch);
  } catch (e) {
    dispatch(removeBookmarkFailure(e.response));
    const { message, type } = getErrorMessage(e);
    dispatchError(message, type, dispatch);
  }
};

export const fetchArticlePage = page => (dispatch, getState) => {
  if (![1, -1].includes(page)) return dispatch(invalidAction());
  const {
    currentPage, size, pagesFetched, totalPages
  } = getState().bookmark;
  const pageToLoad = currentPage + page;
  if (pageToLoad > totalPages || pageToLoad < 1) {
    return dispatch(invalidAction());
  }
  if (pageToLoad > pagesFetched && pagesFetched < totalPages) {
    return dispatch(fetchBookmarkArticle(pageToLoad, size));
  }
  if (pageToLoad <= pagesFetched && pagesFetched <= totalPages) {
    return dispatch(fetchArticlesFromStore(pageToLoad, size));
  }
  return dispatch(invalidAction());
};
