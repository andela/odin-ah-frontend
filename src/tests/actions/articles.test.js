import moxios from 'moxios';
import sinon from 'sinon';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import apiRequest from '../../services/apiRequest';
import * as actions from '../../redux/actions/articles/articles';
import * as types from '../../redux/constants/articles';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({ articles: {} });

const article = {
  title: 'Test title',
  description: 'I am just writing a safety net',
  body: 'I am just writing a safety net',
  tags: ['test']
};

const createSuccessResponse = {
  status: 201,
  data: {
    status: 'success',
    message: 'Article created successfully',
    article
  }
};

const updateSuccessResponse = {
  status: 200,
  data: {
    status: 'success',
    message: 'Article has been updated successfully!',
    article
  }
};

const getArticleSuccessResponse = {
  status: 200,
  data: {
    status: 'success',
    article
  }
};

const errorResponse = {
  status: 500,
  response: {
    status: 'error',
    message: 'Internal server error',
  }
};

describe('Article actions', () => {
  const axiosInstance = apiRequest.getInstance();
  beforeEach(() => {
    store.clearActions();
    moxios.install(axiosInstance);
  });
  afterEach(() => {
    moxios.uninstall(axiosInstance);
  });
  describe('openPublishModal', () => {
    it('should dispatch the OPEN_PUBLISH_MODAL action type', () => {
      store.dispatch(actions.openPublishModal());
      const action = store.getActions();
      expect(action[0]).toEqual({ type: types.OPEN_PUBLISH_MODAL });
      expect(action).toMatchSnapshot();
    });
  });
  describe('closePublishModal', () => {
    it('should dispatch the CLOSE_PUBLISH_MODAL action type', () => {
      store.dispatch(actions.closePublishModal());
      const action = store.getActions();
      expect(action[0]).toEqual({ type: types.CLOSE_PUBLISH_MODAL });
      expect(action).toMatchSnapshot();
    });
  });
  describe('showCreateError', () => {
    const errors = {
      title: ['Title must be at least 5 characters']
    };
    it('should dispatch the SHOW_PUBLISH_ERROR action type', () => {
      store.dispatch(actions.showCreateError(errors));
      const action = store.getActions();
      expect(action[0]).toEqual({ type: types.SHOW_PUBLISH_ERROR, errors });
      expect(action).toMatchSnapshot();
    });
  });
  describe('hideCreateError', () => {
    it('should dispatch the HIDE_PUBLISH_ERROR action type', () => {
      store.dispatch(actions.hideCreateError());
      const action = store.getActions();
      expect(action[0]).toEqual({ type: types.HIDE_PUBLISH_ERROR });
      expect(action).toMatchSnapshot();
    });
  });
  describe('hideCreateResponse', () => {
    it('should dispatch the HIDE_PUBLISH_RESPONSE action type', () => {
      store.dispatch(actions.hideCreateResponse());
      const action = store.getActions();
      expect(action[0]).toEqual({ type: types.HIDE_PUBLISH_RESPONSE });
      expect(action).toMatchSnapshot();
    });
  });
  describe('createArticleRequest', () => {
    it('should dispatch the 2 correct actions', () => {
      moxios.wait(() => {
        const request = moxios.requests.mostRecent();
        request.respondWith(createSuccessResponse);
      });
      return store.dispatch(actions.createArticleRequest(article)).then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
    });

    it('should dispatch the 2 correct error action', () => {
      moxios.wait(() => {
        const request = moxios.requests.mostRecent();
        request.respondWith(errorResponse);
      });
      return store.dispatch(actions.createArticleRequest(article)).then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
    });
  });

  describe('updateArticleRequest', () => {
    const slug = 'test-article-334u4';
    it('should dispatch the 2 correct actions', () => {
      moxios.wait(() => {
        const request = moxios.requests.mostRecent();
        request.respondWith(updateSuccessResponse);
      });
      return store.dispatch(actions.updateArticleRequest(slug, article)).then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
    });

    it('should dispatch the correct error action', () => {
      moxios.wait(() => {
        const request = moxios.requests.mostRecent();
        request.respondWith(errorResponse);
      });
      return store.dispatch(actions.updateArticleRequest(slug, article)).then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
    });
  });

  describe('getArticleForUpdate', () => {
    const slug = 'test-article-334u4';
    it('should dispatch the 2 correct actions', () => {
      moxios.wait(() => {
        const request = moxios.requests.mostRecent();
        request.respondWith(getArticleSuccessResponse);
      });
      return store.dispatch(actions.getArticle(slug)).then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
    });

    it('should dispatch the correct error action', () => {
      moxios.wait(() => {
        const request = moxios.requests.mostRecent();
        request.respondWith(errorResponse);
      });
      return store.dispatch(actions.getArticle(slug)).then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
    });
  });

  async function executeAction(length) {
    await store.dispatch(actions.deleteArticle('url'));
    const action = store.getActions();
    expect(action.length)
      .toEqual(length);
  }
  describe('Comment action test', () => {
    test('should execute getComment action, simulate successful request', async () => {
      const apiReqStub = sinon.stub(apiRequest.axios, 'delete')
        .resolves();
      await executeAction(2);
      apiReqStub.restore();
    });
  });
});
