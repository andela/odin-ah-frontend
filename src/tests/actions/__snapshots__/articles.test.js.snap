// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article actions closePublishModal should dispatch the CLOSE_PUBLISH_MODAL action type 1`] = `
Array [
  Object {
    "type": "CLOSE_PUBLISH_MODAL",
  },
]
`;

exports[`Article actions createArticleRequest should dispatch the 2 correct actions 1`] = `
Array [
  Object {
    "type": "SENDING_REQUEST",
  },
  Object {
    "response": "Article has been published successfully!",
    "type": "CREATE_ARTICLE_SUCCESS",
  },
]
`;

exports[`Article actions createArticleRequest should dispatch the 2 correct error action 1`] = `
Array [
  Object {
    "type": "SENDING_REQUEST",
  },
  Object {
    "response": Object {
      "message": "Internal server error",
      "status": "error",
    },
    "type": "CREATE_ARTICLE_ERROR",
  },
]
`;

exports[`Article actions getArticleForUpdate should dispatch the 2 correct actions 1`] = `
Array [
  Object {
    "type": "SENDING_REQUEST",
  },
  Object {
    "response": undefined,
    "type": "GET_ARTICLE_SUCCESS",
  },
]
`;

exports[`Article actions getArticleForUpdate should dispatch the correct error action 1`] = `
Array [
  Object {
    "type": "SENDING_REQUEST",
  },
  Object {
    "response": Object {
      "message": "Internal server error",
      "status": "error",
    },
    "statusCode": 500,
    "type": "GET_ARTICLE_ERROR",
  },
]
`;

exports[`Article actions hideCreateError should dispatch the HIDE_PUBLISH_ERROR action type 1`] = `
Array [
  Object {
    "type": "HIDE_PUBLISH_ERROR",
  },
]
`;

exports[`Article actions hideCreateResponse should dispatch the HIDE_PUBLISH_RESPONSE action type 1`] = `
Array [
  Object {
    "type": "HIDE_PUBLISH_RESPONSE",
  },
]
`;

exports[`Article actions openPublishModal should dispatch the OPEN_PUBLISH_MODAL action type 1`] = `
Array [
  Object {
    "type": "OPEN_PUBLISH_MODAL",
  },
]
`;

exports[`Article actions showCreateError should dispatch the SHOW_PUBLISH_ERROR action type 1`] = `
Array [
  Object {
    "errors": Object {
      "title": Array [
        "Title must be at least 5 characters",
      ],
    },
    "type": "SHOW_PUBLISH_ERROR",
  },
]
`;

exports[`Article actions updateArticleRequest should dispatch the 2 correct actions 1`] = `
Array [
  Object {
    "type": "SENDING_REQUEST",
  },
  Object {
    "response": "Article has been updated successfully!",
    "type": "UPDATE_ARTICLE_SUCCESS",
  },
]
`;

exports[`Article actions updateArticleRequest should dispatch the correct error action 1`] = `
Array [
  Object {
    "type": "SENDING_REQUEST",
  },
  Object {
    "response": Object {
      "message": "Internal server error",
      "status": "error",
    },
    "type": "UPDATE_ARTICLE_ERROR",
  },
]
`;
